tests/components/sungrow/test_init.py

```diff
@@ -7,27 +7,27 @@
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers import device_registry as dr
 
-from . import MockClient, create_entry, inverter_data
+from . import MockClient, address_variants, create_entry, inverter_data
 
 
 async def test_setup(hass: HomeAssistant) -> None:
-    """Test unload."""
-    entry = create_entry(hass)
-    await hass.config_entries.async_setup(entry.entry_id)
-    await hass.async_block_till_done()
-    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
-    assert entry.state == ConfigEntryState.LOADED
+    """Test creation and unload with different address variants."""
+    for host in address_variants:
+        entry = create_entry(hass, host, 502)
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+        assert entry.state == ConfigEntryState.LOADED
 
-    assert await hass.config_entries.async_unload(entry.entry_id)
-    await hass.async_block_till_done()
+        assert await hass.config_entries.async_unload(entry.entry_id)
+        await hass.async_block_till_done()
 
-    assert entry.state is ConfigEntryState.NOT_LOADED
-    assert not hass.data.get(DOMAIN)
+        assert entry.state is ConfigEntryState.NOT_LOADED
+        assert not hass.data.get(DOMAIN)
```

Comment:
````suggestion
@pytest.mark.parametrize(
    "host",
    [
        "1.1.1.1",
        "http://1.1.1.1",
        "https://1.1.1.1",        
    ],
)
async def test_setup(hass: HomeAssistant, host) -> None:
    """Test creation and unload with different address variants."""
    entry = create_entry(hass, host, 502)
    await hass.config_entries.async_setup(entry.entry_id)
    await hass.async_block_till_done()
    assert entry.state == ConfigEntryState.LOADED

    assert await hass.config_entries.async_unload(entry.entry_id)
    await hass.async_block_till_done()

    assert entry.state is ConfigEntryState.NOT_LOADED
    assert not hass.data.get(DOMAIN)
````

Reply:
`@pytest.mark.parametrize` does this type of test for you.

Reply:
Thank you. Didn't know that.